src/StateService/Verification/VerificationService.cs

```diff
@@ -0,0 +1,158 @@
+using Akka.Actor;
+using Akka.Util.Internal;
+using Neo.IO;
+using Neo.Network.RPC;
+using Neo.Wallets;
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Neo.Plugins.StateService.Verification
+{
+    public class VerificationService : UntypedActor
+    {
+        public class ValidatedRootPersisted { public uint Index; }
+        public class BlockPersisted { public uint Index; }
+        private class Timer { public uint Index; }
+        private static readonly uint TimeoutMilliseconds = ProtocolSettings.Default.MillisecondsPerBlock;
+        private static readonly uint DelayMilliseconds = 3000;
+        private readonly NeoSystem core;
+        private const int MaxCachedVerificationProcessCount = 10;
+        private readonly Wallet wallet;
+        private readonly ConcurrentDictionary<uint, VerificationContext> contexts = new ConcurrentDictionary<uint, VerificationContext>();
+
+        public VerificationService(NeoSystem core, Wallet wallet)
+        {
+            this.core = core;
+            this.wallet = wallet;
+        }
+
+        private void SendVote(VerificationContext context)
+        {
+            var vote = context.CreateVote();
+            if (vote is null) return;
+            Utility.Log(nameof(VerificationService), LogLevel.Info, $"relay vote, height={vote.RootIndex}, retry={context.Retries}");
+            Settings.Default.VerifierUrls.ForEach(url =>
+            {
+                Task.Run(() =>
+                {
+                    try
+                    {
+                        var http_client = new HttpClient
+                        {
+                            Timeout = TimeSpan.FromMilliseconds(TimeoutMilliseconds),
+                        };
+                        var client = new RpcClient(http_client, url);
```

Comment:
We should use `ExtensiblePayload` with p2p, but it could be done in a different PR

Reply:
Yes. Using p2p, We won't need url and avoid inconsistence between urls and public keys.
I have code here https://github.com/ZhangTao1596/neo-modules/tree/state-service-p2p.

@Tommo-L  worries that use p2p will put more pressure on p2p. And he propose add service discovery function.
Once we make agreement, I will commit another pr.

Reply:
If we use service discovery, will it expose the IP address?

Reply:
Can we merge this for now. I will make p2p improvement  pr later.

Reply:
>  use p2p will put more pressure on p2p. 

Yes


> will it expose the IP address?

Maybe we can encrypt part of the service information so that only the relevant node can decrypt it, like:

```
service type + service information

state: xxxxxxx 
oracle: yyyyyyy
rpc: 192.168.1.1
...
```

We can consider it later, after this pr.

Reply:
@Tommo-L Can you create a issue of your idea. We can discuss there.  Since oracle and state both need that.

Reply:
Oracle can't use p2p message for siging!

Reply:
We can discuss here https://github.com/neo-project/neo-modules/issues/493